packages/itinerary-body/src/styled.tsx

```diff
@@ -588,10 +588,14 @@ export const StepsHeaderAndMapLink = styled.span`
   }
 `;
 
-export const StepsHeader = styled(TransparentButton)`
+export const StepsHeader = styled(TransparentButton)<{
+  span?: boolean;
```

Comment:
I would create separate `StepHeaderSpan` and `StepHeaderButton` components and apply the common and specific styles accordingly.